### PR TITLE
React: hook up linking multiple files and make sequencing_date a required field

### LINKED ISSUE #281: Hook up linking multiple files frontend to backend
Spun off from #230

### PR DESCRIPTION
**Pull request by annnag**
Closes #281

### CODE REVIEW

**Review by frewmack — CHANGES_REQUESTED** [review]
The issue is not quite closed since the component used for editing a dataset from the Datasets page is not up-to-date. I'd suggest refactoring the FileLinkingCell component used in DataEntryTable so that the popover is a separate reusable component, then use that in the edit component for the Datasets table.

**Review by kevinlul — COMMENTED** [review]
Add datasets works as expected. I just merged #297, which is the cause of the merge conflicts here

File: react/src/Datasets/components/DatasetTable.tsx [hunk 1]
```diff
@@ -165,15 +164,13 @@ export default function DatasetTable({ isAdmin }: DatasetTableProps) {
...
-                                handleHomeEndKeys
-                                autoHighlight
-                                onChange={(event, newValue) => props.onChange(newValue)}
-                                options={files}
-                                value={props.value}
-                                renderInput={params => <TextField {...params} variant="standard" />}
+                            <FileLinkingComponent
+                                values={props.rowData.linked_files}
+                                options={files.map(
+                                    file => ({ title: file, inputValue: file } as Option)
```
Comment (kevinlul) [inline react/src/Datasets/components/DatasetTable.tsx:170]: Is the typecast here needed?

File: react/src/components/FileLinkingComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+                        color="default"
+                        size="small"
+                        onClick={handleClick}
+                        disabled={props.disabled}
+                        disableRipple
+                        disableElevation
+                    >
+                        {props.values.length} files
+                    </Button>
+                </span>
```
Comment (kevinlul) [inline react/src/components/FileLinkingComponent.tsx:160]: Same question here

**Review by kevinlul — CHANGES_REQUESTED** [review]

File: react/src/AddDatasets/components/TableCells.tsx [hunk 3]
```diff
@@ -507,13 +226,15 @@ export function DataEntryActionCell(props: {
...
-                </IconButton>
+                <span>
+                    <IconButton
+                        onClick={props.onClick}
+                        disabled={props.disabled}
+                        aria-label={props.tooltipTitle}
+                    >
+                        {props.icon}
+                    </IconButton>
+                </span>
```
[the remark this answers is not in the document]
Comment (annnag) [reply react/src/AddDatasets/components/TableCells.tsx:237]: To fix this error in the console:`You are providing a disabled button child to the Tooltip component`. This wrapper element also allows the tooltip to be activated when the button is disabled.

**Review by kevinlul — APPROVED** [review]

**Pull request merged**